api/src/multiaccessy/accessy.py

```diff
@@ -42,107 +43,215 @@ def create_session(cls, client_id: str, client_secret: str) -> "AccessySession":
+            if len(current_items) == 0 and total_item_count != 0:
+                raise ValueError(f"Could not get all items. Not enough items were returned from Accessy endpoint during pagination loop. Got {received_item_count} out of {total_item_count} expected items")
 
         return items
+    
+    ################################################
+    # Methods that return raw JSON data from API:s
+    ################################################
 
-    def _get_organization(self) -> str:
-        response = requests.get(ACCESSY_URL + "/asset/user/organization-membership",
-                                headers={"Authorization": f"Bearer {self.session_token}"})
-        # [{"id":<uuid>,"userId":<uuid>,"organizationId":<uuid>,"roles":[<roles>]}]
-
-        check_response_error(response, "Get organization")
-
-        data = response.json()
-        if len(data) > 1:
-            warnings.warn("API key has several memberships. This is probably an error...", RuntimeWarning)
-        elif len(data) == 0:
-            raise ValueError("The API key does not have a corresponding organization membership")
-
-        return data[0]["organizationId"]
+    def _get_user_details(self, user_id: UUID) -> dict:
+        """ Get details for user ID. Fields: id, msisdn, firstName, lastName, ... """
+        return self._get_json(f"/org/admin/user/{user_id}", msg="Getting user details")
 
-    def get_users(self) -> list[dict]:
+    def _get_users_org(self) -> list[dict]:
+        """ Get all user ID:s """
         return self._get_json_paginated(f"/asset/admin/organization/{self.organization_id}/user")
         # {"items":[{"id":<uuid>,"msisdn":"+46...","firstName":str,"lastName":str}, ...],"totalItems":6,"pageSize":25,"pageNumber":0,"totalPages":1}
+    
+    def _get_users_in_access_group(self, access_group_id: UUID) -> list[dict]:
+        """ Get all user ID:s in a specific access group """
+        return self._get_json_paginated(f"/asset/admin/access-permission-group/{access_group_id}/membership")
+        # {"items":[{"id":<uuid>,"userId":<uuid>,"organizationId":<uuid>,"roles":[<roles>]}, ...],"totalItems":3,"pageSize":25,"pageNumber":0,"totalPages":1}
+
+    def _get_users_lab(self) -> list[dict]:
+        """ Get all user ID:s with lab access """
+        return self._get_users_in_access_group(ACCESSY_LAB_ACCESS_GROUP)
+
+    def _get_users_special(self) -> list[dict]:
+        """ Get all user ID:s with special access """
+        return self._get_users_in_access_group(ACCESSY_SPECIAL_ACCESS_GROUP)
+    
+    def _user_ids_to_accessy_members(self, user_ids: list[UUID]) -> list["AccessyMember"]:
+        """ Convert a list of User ID:s to AccessyMembers """
+
+        APPLICATION_PHONE_NUMBER = object()  # Sentinel phone number for applications
+        def fill_user_details(user: "AccessyMember", user_id: str):
+            data = self._get_json(f"/org/admin/user/{user_id}")
+
+            # API keys do not have phone numbers
```

Comment:
Improve comment about why this is needed?

Reply:
True, updated the comment :)